Mage.Server/src/main/java/mage/server/util/ServerConnection.java

```diff
@@ -0,0 +1,96 @@
+package mage.server.util;
+
+import mage.server.util.config.Server;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.log4j.Logger;
+
+import java.net.*;
+import java.util.Enumeration;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class ServerConnection {
```

Comment:
This is an alternate to the `Connection` which was originally shared with the client. This class support the legacy connection `getUri` behaviour (with calculation of the bind address from the interface) but also support the multi-home connection depending on the value of the configuration. The legacy behaviour is just copy-pasted from the `Connection` class to prevent regressions on the client side.